### PR TITLE
Drop tls connection when tls error occurred instead of bubble up error

### PR DESCRIPTION
**Pull request by dmolokanov**
When an error occurred on TLS handshake HTTP server erroneously reported an error and exit stopped program execution. 
Now it ignores this type of errors and continues to execute the next available requests.

### CODE REVIEW

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 1]
```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
...
-                    .iter_mut()
-                    .map(|(fut, _)| fut.poll())
-                    .enumerate()
-                    .find(|(_, result)| match result {
-                        Ok(v) => v.is_ready(),
-                        Err(_) => true,
-                    });
+                loop {
+                    // Look through the connections list for the first connection that is either ready to be
+                    // passed to the stream selector or has failed and needs the error bubbled up.
```
Comment (arsing) [inline edgelet/edgelet-http/src/util/incoming.rs]: "or has failed and needs the error bubbled up" is outdated now.

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 2]
```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
...
-                                #[cfg(windows)]
-                                {
-                                    Async::Ready(Some(
-                                        (Box::new(StreamSelector::Tls(tls_stream), addr)),
-                                    ))
+                                // The prior block included a filter that specifically asked for is_ready state,
+                                // so this line is unreachable.
+                                Ok(_) => unreachable!(),
+                                Err(err) => {
+                                    log_failure(Level::Warn, &err);
```
Comment (arsing) [inline edgelet/edgelet-http/src/util/incoming.rs]: Maybe add a comment like `// Ignore TLS handshake errors.`

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 3]
```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
...
                                 }
-                                #[cfg(windows)]
-                                {
-                                    Async::Ready(Some(
-                                        (Box::new(StreamSelector::Tls(tls_stream), addr)),
-                                    ))
+                                // The prior block included a filter that specifically asked for is_ready state,
+                                // so this line is unreachable.
+                                Ok(_) => unreachable!(),
+                                Err(err) => {
```
Comment (kkdawkins) [inline edgelet/edgelet-http/src/util/incoming.rs]: Is there any error filtering that we want to do or just drop connection on any/all errors?

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 4]
```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
...
-                // Return a tuple containing the index and state.
-                let val = connections
-                    .iter_mut()
-                    .map(|(fut, _)| fut.poll())
-                    .enumerate()
-                    .find(|(_, result)| match result {
-                        Ok(v) => v.is_ready(),
-                        Err(_) => true,
-                    });
+                loop {
```
Comment (kkdawkins) [inline edgelet/edgelet-http/src/util/incoming.rs]: You made this a loop?

File: edgelet/edgelet-http/src/util/incoming.rs (hunk 3 above)
Comment (dmolokanov) [reply edgelet/edgelet-http/src/util/incoming.rs]: It rather complicated to implement (I don't want to say impossible), but
1. This error type doesn't support ErrorKind - difficult to filter out the error
2. This is an error that comes from native_tls, so I assume that it is the type we are interested in.

During a quick discussion with @arsing we decided that it is good enough to leave like this.

File: edgelet/edgelet-http/src/util/incoming.rs (hunk 4 above)
Comment (dmolokanov) [reply edgelet/edgelet-http/src/util/incoming.rs]: Yes. This loop covers the situation when we have several items in a `vec` that are ready to process. Imagine, if one of them has an error, and another doesn't have. We process the first one and return `Async::NotReady`. But we have another one that has no task to signal that future is ready again. 
Other words, once we started processing inside the `poll` method we should process all futures that are ready to poll.

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 5]
```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
...
+                            match result {
+                                Ok(Async::Ready(tls_stream)) => {
+                                    #[cfg(not(windows))]
+                                    {
+                                        return Ok(Async::Ready(Some((
+                                            StreamSelector::Tls(tls_stream),
+                                            addr,
+                                        ))));
+                                    }
+                                    #[cfg(windows)]
```
Comment (dmolokanov) [inline edgelet/edgelet-http/src/util/incoming.rs]: @kkdawkins Why do we need this block for windows? I thought whole TLS thing is for [unix only](https://github.com/Azure/iotedge/blob/a11c02fb733c576853b609c387a81faeab11ab0b/edgelet/edgelet-http/src/util/incoming.rs#L55)

File: edgelet/edgelet-http/src/util/incoming.rs [hunk 6]
```diff
@@ -5,8 +5,7 @@ use std::io;
 use std::sync::Mutex;
 
 use futures::{Poll, Stream};
-#[cfg(unix)]
-use tokio::io::{Error as TokioIoError, ErrorKind as TokioIoErrorKind};
+use log::Level;
```
Comment (arsing) [inline edgelet/edgelet-http/src/util/incoming.rs]: As the CI failure says, this import and the `log_failure` one below need to be `#[cfg(unix)]`

Comment (dmolokanov) [reply edgelet/edgelet-http/src/util/incoming.rs]: Yeah, I noticed that. Thanks!

**Pull request merged**